File: LeetCode/344 Reverse String/main.cpp
#include <iostream>

using namespace std;

class Solution {
public:
    void reverseString(vector<char> & s) {
        for (int first = 0, last = s.size()-1; first<last; first++, last--)
            iter_swap(s.begin() + first, s.begin() + last);
    }
};

File: LeetCode/344 Reverse String/main.js
/**
 * @param {character[]} s
 * @return {void} Do not return anything, modify s in-place instead.
 */
var reverseString = function(s) {
    /**
     * @param {character[]} s: stringa da ruotare
     * @param {int} left: estremo sinistro dell'array da ruotare
     * @param {int} right: estremo destro dell'array da ruotare (non compreso)
     */
    var reverse = (s, left = 0, right = s.length) => {
        if (left < 0 || right < 0)
            return;

        for (var first = left, last = right-1; first < last; first++, last--){
            var tmp = s[last];
            s[last] = s[first];
            s[first] = tmp;
        }
    }

    reverse(s)
};